core/unit_test/TestViewMapping_a.hpp

```diff
@@ -671,11 +668,13 @@ void test_view_mapping() {
     ASSERT_TRUE((std::is_same_v<typename T::const_data_type, const int*>));
     ASSERT_TRUE((std::is_same_v<typename T::non_const_data_type, int*>));
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
     ASSERT_TRUE((std::is_same_v<typename T::scalar_array_type, int*>));
```

Comment:
Why aren't we getting deprecated warnings and forced to `KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_{POP,PUSH}()`?